### PR TITLE
chore: update supported python versions in README to >= 3.6

### PR DESCRIPTION
**Pull request by rajatbhatta**
Update supported python versions in README to >= 3.6, in line with setup.py which indicates python_requires=">=3.6".

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -56,7 +56,7 @@ dependencies.
 
 Supported Python Versions
 ^^^^^^^^^^^^^^^^^^^^^^^^^
-Python >= 3.5
+Python >= 3.6
```
Comment (ansh0l) [inline README.rst:59]: Would this mean all customers using python 3.5 face incompatibility with library?

Comment (rajatbhatta) [reply README.rst:59]: I think so. 

Support for Python 3.5 and earlier versions was removed as part of this commit: https://github.com/googleapis/python-spanner/pull/147/commits/606258bc728e5eed2b0b032a6428e1b77c39b3ce.

Adding @vi3k6i5 to comment/confirm the same.

Comment (IlyaFaer) [reply README.rst:59]: Users will get problems if they'll try to upgrade to the new version of the `python-spanner` package. See the setup param for details:
https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#python-requires

So, until they'll try to upgrade, they're okay.
Plus to this, I don't really think it's a big move to upgrade from 3.5 to 3.6. It's not jumping from 2 to 3, it's a comparatively small step, which, most likely, will not require manual work.

Comment (vi3k6i5) [reply README.rst:59]: Makes sense. This is anyways just a documentation fix, the setup.py file was updated a while back.

Comment (vi3k6i5) [reply README.rst:59]: Maybe we should add Python 3.5 to Deprecated Python Versions below ? @ansh0l @IlyaFaer ?

**Review by vi3k6i5 — APPROVED** [review]
LGTM

File: README.rst (hunk 1 above)
Comment (IlyaFaer) [reply README.rst:59]: Yeah, I guess, let's add

Comment (rajatbhatta) [reply README.rst:59]: This is done.

Comment (ansh0l) [reply README.rst:59]: sounds good

**Review by ansh0l — APPROVED** [review]
lgtm

**Pull request merged**